### ISSUE TITLE
Database structure documentation

### ISSUE THREAD

**Issue by scramatte** [ZakariaTlilani/nova-nested-tree#2]
Hi,

I'm trying to use your field in my project, but I've got some difficulties with database structure.
Can you add a complete example? I mean with migrations and models.

In my case, I've got  products and product_famillies tables
I don't know which should be the relation between my famillies  tree and product table
Do you store selected categories as JSON or using pivot table?

--product
id
name
...
familly ?

--product_famillies
_lft
_rgt
parent_id
name
..

Thank you for the help
Regards

**Comment by scramatte** [comment]
Hello,

Thank you to reply so quickly.  If category id is int you can only choose only one category.
As I need to store various,  I've created a pivot table with belongsToMany relationship.

Now, I've been able to store famillies info but when I reload page field is empty. 
Any idea?

Regards

**Comment by scramatte** [comment]
I've tried with options ->useAsField() and ->useSingleSelect  but I get the same error

**Comment by ZakariaTlilani** [comment]
The issue is in the relationship, that's's why the JSON return in not found, please share your models (product and Product_families) / nova resource

**Comment by ZakariaTlilani** [comment]
see this simple example: 

![Screenshot from 2024-05-01 14-55-04](https://github.com/ZakariaTlilani/nova-nested-tree/assets/116965173/e4fe06bc-5726-4428-89c0-8a4a067be699)
![Screenshot from 2024-05-01 14-55-39](https://github.com/ZakariaTlilani/nova-nested-tree/assets/116965173/024d9cce-de08-435c-8b08-648ddb3bc257)

This example is to select subcategory mother category (parent), there is a relationship called "parent" wish we use in the nova resource as a parameter. Please make sure your database in set Wright as explained before.
it is as simple as this.

**Comment by scramatte** [comment]
Hi,

It still doesn't works ... 
If I use a native Nova BelongsTo relation it's working
![imagen](https://github.com/ZakariaTlilani/nova-nested-tree/assets/188766/d3e30aa6-76e7-4049-9076-8f3e2deed369)

Model
```php
<?php

namespace Modules\Billing\Models\Product;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Product extends Model
{
    protected $table = 'billing.products';

    public function family(): BelongsTo {
        return $this->belongsTo(ProductFamily::class);
    }

}
```
Nova Resource
```php
<?php

namespace Modules\Billing\Nova\Resources\Product;

use Illuminate\Http\Request;

use Laravel\Nova\Fields\Text;
use Laravel\Nova\Fields\BelongsTo;
use ZakariaTlilani\NovaNestedTree\NestedTreeAttachManyField;

use Modules\Billing\Models\Product as ProductModels;
use Modules\Billing\Nova\Resources\Resource;

use Modules\Billing\Nova\Resources as BillingResources;

class Product extends Resource
{
    /**
     * The model the resource corresponds to.
     *
     * @var string
     */
    public static $model = ProductModels\Product::class;

    /**
     * The single value that should be used to represent the resource when being displayed.
     *
     * @var string
     */
    public static $title = 'name';

    /**
     * The columns that should be searched.
     *
     * @var array
     */
    public static $search = [
        'id',
    ];

    /**
     * Get the fields displayed by the resource.
     *
     * @return array
     */
    public function fields(Request $request)
    {
        return [
            BelongsTo::make(__('billing.product_family'),'family',ProductFamily::class),
            NestedTreeAttachManyField::make(__('billing.product_family'),'family',ProductFamily::class)
                ->useSingleSelect(),
            Text::make(__('billing.product_name'), 'name'),
        ];
    }

    /**
     * Get the cards available for the request.
     *
     * @return array
     */
    public function cards(Request $request)
    {
        return [];
    }

    /**
     * Get the lenses available for the resource.
     *
     * @return array
     */
    public function lenses(Request $request)
    {
        return [];
    }

    /**
     * Get the actions available for the resource.
     *
     * @return array
     */
    public function actions(Request $request)
    {
        return [];
    }
}
```

**Comment by scramatte** [comment]
The problem is not the relation, it looks that the HTTP controller doesn't exists 

http:///localhost/nova-vendor/nova-nested-tree-attach-many/products/20/attached/family/id

**Comment by ZakariaTlilani** [comment]
I see no issue in your code, verify your database.
If you debug your code successfully, you will get a JSON return of your "families".
This from production:
![Untitled](https://github.com/ZakariaTlilani/nova-nested-tree/assets/116965173/09d70d38-d18b-4cd9-a7d3-02a9fc85c7c0)

**Comment by scramatte** [comment]
I've just submit a PR to fix these issues (#5)
